app/redux/sagas/SyncSaga.js

```diff
@@ -67,7 +68,13 @@ function* pushUpdates() {
             // upload
             const initialRevision = yield select(SetsSelectors.getRevision);            
             const accessToken = yield select(AuthSelectors.getAccessToken);
-            const json = yield call(API.postUpdatedSetData, sets, accessToken);
+            const lastRefreshDate = yield select(AuthSelectors.lastRefreshDate);            
+            const validator = {
+                accessToken,
+                lastRefreshDate,
+                isValid: Validator.validator
+            };    
```

Comment:
Instead of making a validator object here that contains another validator...

...move this all into Validator.js. Validator.js should contain the tokens and refresh dates.